sandbox-app/on-change.tsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import { useForm } from '../src';
+
+export function OnChange(): JSX.Element {
+  const data = React.useRef({
+    name: 'John Doe',
+    confirm: false,
+  });
+
+  const { fields, changes, touched, onSubmit, reset } = useForm({
+    model: data.current,
+    onSubmit: async ({ model }) => {
+      data.current = model;
+    },
+    onChange: async ({ changes, touched }) => {
+      data.current = {
+        ...data.current,
+        ...changes,
+        ...touched,
+      };
+    },
+  });
+
+  return (
+    <>
+      <h2 id="on-change">On Change</h2>
+      <p>
+        Changes does not reflect a double toggled boolean state because the
+        value will be the same as the initial value. We can now use the touched
+        object to get an updated model with the changes.
+      </p>
+
+      <form onSubmit={onSubmit}>
+        <label>
+          <span>Confirm</span>
+          <input
+            type="checkbox"
+            checked={fields.confirm.value}
+            onChange={(event) => {
+              fields.confirm.touch();
```

Comment:
why is this triggered manually? shouldn't the form lib set it automatically on change?

Reply:
touched is generally triggered onBlur. 

In a real case scenario we don't need this because we pass the whole ...field object to our custom form fields

Reply:
could we do that here as well to reflect real life usage?

Reply:
We decided in the past not to set touched in the onChange handler because we usually show the errors when the field is touched.

This would mean you would show an invalid state as the user is interacting with the field and not after they finished typing.

Reply:
> could we do that here as well to reflect real life usage?

I think this works better for the sandbox.